SpatialGDK/Source/SpatialGDK/Public/Utils/RPCContainer.h

```diff
@@ -6,20 +6,81 @@
 #include "Schema/UnrealObjectRef.h"
 #include "SpatialConstants.h"
 
+#include "UObject/Object.h"
+#include "UObject/Class.h"
+#include "UObject/WeakObjectPtr.h"
 #include "CoreMinimal.h"
 
+DECLARE_LOG_CATEGORY_EXTERN(LogRPCContainer, Log, All);
+
 struct FPendingRPCParams;
-using FPendingRPCParamsPtr = TUniquePtr<FPendingRPCParams>;
-DECLARE_DELEGATE_RetVal_OneParam(bool, FProcessRPCDelegate, const FPendingRPCParams&)
+struct FRPCErrorInfo;
+DECLARE_DELEGATE_RetVal_OneParam(FRPCErrorInfo, FProcessRPCDelegate, const FPendingRPCParams&)
 
-struct SPATIALGDK_API FPendingRPCParams
+enum class ERPCError : uint8_t
+{
+	Success,
+
+	// Shared across Sender and Receiver
+	NoProcessingFunctionBound,
+	UnresolvedTargetObject,
+	MissingFunctionInfo,
+	UnresolvedParameters,
+
+	// Sender specific
+	NoActorChannel,
+	SpatialActorChannelNotListening,
+	NoNetConnection,
+	NoAuthority,
+	InvalidRPCType,
+
+	// Specific to packing
+	NoOwningController,
+	NoControllerChannel,
+	UnresolvedController,
+	ControllerChannelNotListening,
+
+	Unknown
+};
+
+enum class ERPCQueueType : uint8_t
+{
+	Send,
+	Receive,
+	Unknown
+};
+
+struct FRPCErrorInfo
 {
-	FPendingRPCParams(const FUnrealObjectRef& InTargetObjectRef, SpatialGDK::RPCPayload&& InPayload, int InReliableRPCIndex = 0);
+	bool Success() const
+	{
+		return (ErrorCode == ERPCError::Success);
+	}
+
+	TWeakObjectPtr<UObject> TargetObject;
+	TWeakObjectPtr<UFunction> Function;
+	bool bIsServer;
+	ERPCQueueType QueueType;
```

Comment:
Should either of these enums have default values?

Reply:
Added default values.